### ISSUE TITLE
Improve class extendability

### ISSUE THREAD

**Issue by petski** [fschmtt/keycloak-rest-api-client-php#128]
For several reasons, I extend the `Fschmtt\Keycloak\Keycloak` class. This class has all it's properties and the `fetchVersion()` method set to `private`. 

Please update to `protected` so I can inherit and access these properties and method.

I'd be happy to supply a PR if you're open for this change.

**Comment by fschmtt** [comment]
Hey, thanks you for opening this issue.

Could you please elaborate your use case? Maybe we can find another solution that allows extending the library as you need without making these properties `protected` 😄

**Comment by petski** [comment]
Thanks for getting back to me so quick. 

My use-case is that I'm using Organizations (as you may have noticed :smiley:). At least two of the methods for Organizations expects form data, instead of JSON.

See https://github.com/keycloak/keycloak/blob/26.0.0/js/libs/keycloak-admin-client/src/resources/organizations.ts#L107 for the methods using form data. I.e. `/admin/realms/{realm}/organizations/{orgId}/members/invite-user` and `/admin/realms/{realm}/organizations/{orgId}/members/invite-existing-user`.

Your project currently forces JSON data to be send to Keycloak. Unfortunately, in my (custom) Organizations resource, I cannot access the client in any way to implement the form data approach. So I've tried to extend from your class, but couldn't get further because the client was private ...



FWIW, currently my extended client looks like this:

```php
<?php

declare(strict_types=1);

namespace App\Security\Keycloak;

use App\Security\Keycloak\Resource\Organizations;
use Fschmtt\Keycloak\Http\Criteria;
use Fschmtt\Keycloak\Http\Method;
use Fschmtt\Keycloak\Keycloak as VendorKeycloakClient;
use Fschmtt\Keycloak\Representation\Organization as KeycloakOrganization;
use Fschmtt\Keycloak\Representation\User as KeycloakUser;
use RuntimeException;

class AdminClient extends VendorKeycloakClient
{
    /**
     * @param array<string, string> $seachConditions
     */
    public function findOneUserBy(string $realm, array $seachConditions): ?KeycloakUser
    {
        $collection = $this->users()->search($realm, new Criteria($seachConditions));

        if ($collection->count() > 1) {
            throw new RuntimeException('Multiple users found');
        }

        return $collection->first();
    }

    /**
     * @param array<string, string> $seachConditions
     */
    public function findOneOrganizationBy(string $realm, array $seachConditions): ?KeycloakOrganization
    {
        $collection = $this->organizations()->search($realm, new Criteria($seachConditions));

        if ($collection->count() > 1) {
            throw new RuntimeException('Multiple organizations found');
        }

        return $collection->first();
    }

    /**
     * @see https://github.com/keycloak/keycloak/blob/26.0.0/js/libs/keycloak-admin-client/src/resources/organizations.ts#L107
     *
     * Implemented here, and not in a resource class, because the resource class forces json, where we need form data.
     */
    public function inviteUser(string $realm, string $orgId, string $email, ?string $firstName = null, ?string $lastName = null): void
    {
        $this->client->request(
            Method::POST->value,
            sprintf('/admin/realms/%s/organizations/%s/members/invite-user', $realm, $orgId),
            [
                'form_params' => [
                    'email'     => $email,
                    'firstName' => $firstName,
                    'lastName'  => $lastName,
                ],
            ],
        );
    }

    public function organizations(): Organizations
    {
        /** @var Organizations */
        return $this->resource(Organizations::class);
    }
}
```

**Comment by fschmtt** [comment]
Thanks for clarification!

Unfortunately there seems to be no documentation for this endpoint (or any other `organizations` endpoint) on the official [Keycloak Administration REST API](https://www.keycloak.org/docs-api/latest/rest-api/index.html) docs.

Right now I think it would make sense to think about changing the `CommandExecutor` to allow `application/x-www-form-urlencoded` to be sent instead of only JSON payloads.

Can you provide example HTTP requests to the endpoint?

**Comment by petski** [comment]
Also noticed it isn't in the official REST documentation yet, but it is part of Keycloak's functionality:

https://github.com/keycloak/keycloak/blob/26.0.0/services/src/main/java/org/keycloak/organization/admin/resource/OrganizationMemberResource.java#L115

Would this be enough for now, or do you really need the HTTP requests? I can provide those somewhere later this week if desired.

**Comment by petski** [comment]
This is an example HTTP request:

![image](https://github.com/user-attachments/assets/5c5cbdd0-33be-461c-8065-ff6f69dfdc7f)

You can set `firstName` and `lastName` to `null` (or even leave the keys out).

If **BOTH** `firstName` and `lastName` are provided, then `orgInviteBodyPersonalized` and `orgInviteBodyPersonalizedHtml` are used in the invite email.

If only `email` is provided, then `orgInviteBody` and `orgInviteBodyHtml` are used

**Comment by fschmtt** [comment]
Thanks for all the information!

Please allow me some time to get back to this, I'm quite busy recently, sorry :/

**Comment by fschmtt** [comment]
Hey, thank you for your patience and sorry for not replying earlier 🙈 

I made a change to the `CommandExecutor` to send `form_params` if its `$payload` property is an `array`. You can find and try out the changes here: https://github.com/fschmtt/keycloak-rest-api-client-php/pull/136

I could imagine that this change might break other endpoints that still requires a JSON payload.

I would appreciate if you could have a look at the change(s), try them out and give me some feedback 🙂 Thank you!

**Comment by petski** [comment]
Hi @fschmtt, thanks for your work in #136.

I'm happy to confirm it works a desired. I tested with keycloak 26.0.7.

The only feedback I have is that `Organizations.php` could use a `public function create(..)` as `Users`, `Groups` and `Roles` do have. I added this to my own `Organizations`-class (which extends from yours), with:

```php
/**
 * @see https://www.keycloak.org/docs-api/26.0.7/rest-api/index.html#_post_adminrealmsrealmorganizations
 */
public function create(string $realm, OrganizationRepresentation $organization): void
{
    $this->commandExecutor->executeCommand(
        new Command(
            '/admin/realms/{realm}/organizations',
            Method::POST,
            [
                'realm' => $realm,
            ],
            $organization,
        ),
    );
}
```

Could also something to add later, of course.

**Comment by fschmtt** [comment]
I got back to it and released v0.29.0 including the organizations 😄
